ArduCopter/toy_mode.cpp

```diff
@@ -1080,4 +1096,70 @@ void ToyMode::arm_check_compass(void)
     }
 }
 
+/*
+  handle take of cmd
+ */
+void ToyMode::handle_takeoff_cmd(bool takeoff_btn,bool takeoff_cancel)
+{
+    const uint32_t takeoff_delay_ms = 3000;
+    
+    //initiate takeoff cmd
+    if (takeoff_state == TAKEOFF_ON_GROUND && !copter.motors->armed() && takeoff_btn) {
+        takeoff_state = TAKEOFF_INITIATE;
+        takeoff_last_press_ms = AP_HAL::millis();
+    }
+    
+    switch (takeoff_state) {
+    case TAKEOFF_INITIATE:
+        
+        if (!copter.motors->armed()) {
+            action_arm();
```

Comment:
I got one here in the same state - https://github.com/SkyRocketToys/ardupilot/blob/czar-launch-cmd/ArduCopter/toy_mode.cpp#L1129

wont this keep trying until its armed then it moves to TAKEOFF_ARMED state. 

But come to think about it I also need to consider other scenario that would cause it to fail to arm i.e. copter is upside down or even wait until everything is initialised properly before moving to the next state.  

At the moment I only have one that supports a transition to alt_hold when in loiter and no GPS. - https://github.com/SkyRocketToys/ardupilot/blob/czar-launch-cmd/ArduCopter/toy_mode.cpp#L1123